gravitee-rest-api-service/src/main/java/io/gravitee/rest/api/service/impl/ApiDuplicatorServiceImpl.java

```diff
@@ -0,0 +1,737 @@
+                        planService.create(newPlanEntity);
+                    }
+                } else {
+                    PlanQuery query = new PlanQuery.Builder()
+                        .api(createdOrUpdatedApiEntity.getId())
+                        .name(planNode.get("name").asText())
+                        .security(PlanSecurityType.valueOf(planNode.get("security").asText().toUpperCase()))
+                        .build();
+                    List<PlanEntity> planEntities = planService
+                        .search(query)
+                        .stream()
+                        .filter(planEntity -> !PlanStatus.CLOSED.equals(planEntity.getStatus()))
+                        .collect(toList());
+                    if (planEntities.isEmpty()) {
+                        NewPlanEntity newPlanEntity = objectMapper.readValue(planNode.toString(), NewPlanEntity.class);
+                        newPlanEntity.setApi(createdOrUpdatedApiEntity.getId());
+                        planService.create(newPlanEntity);
+                    } else if (planEntities.size() == 1) {
+                        UpdatePlanEntity updatePlanEntity = objectMapper.readValue(planNode.toString(), UpdatePlanEntity.class);
+                        updatePlanEntity.setId(planEntities.iterator().next().getId());
+                        planService.update(updatePlanEntity);
+                    } else {
+                        LOGGER.error(
+                            "Not able to identify the plan to update: {}. Too much plan with the same name",
+                            planNode.get("name").asText()
+                        );
+                        throw new TechnicalManagementException(
+                            "Not able to identify the plan to update: " +
+                            planNode.get("name").asText() +
+                            ". Too much plan with the same name"
+                        );
+                    }
+                }
+            }
+        }
+        // Metadata
+        final JsonNode metadataDefinition = jsonNode.path("metadata");
+        if (metadataDefinition != null && metadataDefinition.isArray()) {
+            try {
+                for (JsonNode metadataNode : metadataDefinition) {
+                    UpdateApiMetadataEntity updateApiMetadataEntity = objectMapper.readValue(
+                        metadataNode.toString(),
+                        UpdateApiMetadataEntity.class
+                    );
+                    updateApiMetadataEntity.setApiId(createdOrUpdatedApiEntity.getId());
+                    apiMetadataService.update(updateApiMetadataEntity);
+                }
+            } catch (Exception ex) {
+                LOGGER.error("An error occurs while creating API metadata", ex);
+                throw new TechnicalManagementException("An error occurs while creating API Metadata", ex);
+            }
+        }
+    }
+
+    private static class MemberToImport {
```

Comment:
do you think we can extract this to a dedicated file ?

Reply:
Discussed, we will do it in another PR because it needs some extra refactoring.